packages/world/src/modules/core/implementations/WorldRegistrationSystem.sol

```diff
@@ -174,6 +174,19 @@ contract WorldRegistrationSystem is System, IWorldErrors {
     FunctionSelectors.set(worldFunctionSelector, resourceSelector, systemFunctionSelector);
   }
 
+  /**
+   * Register World function selectors in batch for the given namespace, name and system function.
+   */
+  function registerFunctionSelectors(
+    bytes32 resourceSelector,
+    string[] calldata systemFunctionNames,
+    string[] calldata systemFunctionArguments
```

Comment:
would `calldata` here allow for calling this on-chain? I thought that the reason for `memory` was to allow for both on-chain calls and off-chain calls

Reply:
OK Idk we need to call it on-chain. I have updated it to `memory` now